# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureClientSharedSurface.h
# Commit: 204b899f436d
# Full Hash: 204b899f436d99f0eae759af3ccfcbaabfbbb428
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 3f5baab34b1f -r 204b899f436d gfx/layers/client/TextureClientSharedSurface.h
--- a/gfx/layers/client/TextureClientSharedSurface.h	Thu Jun 11 06:36:55 2020 +0000
+++ b/gfx/layers/client/TextureClientSharedSurface.h	Thu Jun 11 06:37:35 2020 +0000
@@ -30,14 +30,15 @@
 class SharedSurfaceTextureClient;
 
 class SharedSurfaceTextureData : public TextureData {
- protected:
-  const UniquePtr<gl::SharedSurface> mSurf;
-
   friend class SharedSurfaceTextureClient;
 
-  explicit SharedSurfaceTextureData(UniquePtr<gl::SharedSurface> surf);
+ public:
+  const SurfaceDescriptor mDesc;
+  const gfx::SurfaceFormat mFormat;
+  const gfx::IntSize mSize;
 
- public:
+  SharedSurfaceTextureData(const SurfaceDescriptor&, gfx::SurfaceFormat,
+                           gfx::IntSize);
   virtual ~SharedSurfaceTextureData();
 
   bool Lock(OpenMode) override { return false; }
@@ -49,10 +50,8 @@
   bool Serialize(SurfaceDescriptor& aOutDescriptor) override;
 
   void Deallocate(LayersIPCChannel*) override;
-
-  gl::SharedSurface* Surf() const { return mSurf.get(); }
 };
-
+/*
 class SharedSurfaceTextureClient : public TextureClient {
  public:
   SharedSurfaceTextureClient(SharedSurfaceTextureData* aData,
@@ -60,16 +59,16 @@
 
   ~SharedSurfaceTextureClient();
 
-  static already_AddRefed<SharedSurfaceTextureClient> Create(
+  static RefPtr<SharedSurfaceTextureClient> Create(
       UniquePtr<gl::SharedSurface> surf, gl::SurfaceFactory* factory,
       LayersIPCChannel* aAllocator, TextureFlags aFlags);
 
-  gl::SharedSurface* Surf() const {
+  const auto& Desc() const {
     return static_cast<const SharedSurfaceTextureData*>(GetInternalData())
-        ->Surf();
+        ->mDesc;
   }
 };
-
+*/
 }  // namespace layers
 }  // namespace mozilla
 